server/src-lib/Hasura/RQL/DDL/RemoteRelationship/Validate.hs

```diff
@@ -407,16 +407,18 @@ validateType permittedVariables value expectedGType types =
 
 assertType :: G.GType -> G.GType -> Validation (NonEmpty ValidationError) ()
 assertType actualType expectedType = do
-  -- check if both are list types or both are named types
-  (when
-     (isListType' actualType /= isListType' expectedType)
-     (Failure (pure $ ExpectedTypeButGot expectedType actualType)))
-  -- if list type then check over unwrapped type, else check base types
-  if isListType' actualType
-    then assertType (unwrapTy actualType) (unwrapTy expectedType)
-    else (when
-            (getBaseTy actualType /= getBaseTy expectedType)
-            (Failure (pure $ ExpectedTypeButGot expectedType actualType)))
+  -- The GraphQL spec says that, we can coerce a singleton value into an array
```

Comment:
This comment is great. While you are here can you give `assertType` a better name? I guess what we're actually doing here is asserting that `actualType` is coercible to `expectedType`?

Reply:
Also, is this deficient in other ways according to the spec?

Reply:
> I guess what we're actually doing here is asserting that actualType is coercible to expectedType?

Yes, that's what we're doing. But I'm not sure, what can be a better name here? Perhaps, `isTypeCoercible`?

Reply:
> Also, is this deficient in other ways according to the spec?

Yes, it was. My patch would have allowed to make a remote relationship joining types `[[Int]]` and `[[[Int]]]`, which is illegal, as specified in the GraphQL spec.

I have fixed this in the next commit

Reply:
`isTypeCoercible` sounds good to me!